### ISSUE TITLE
FileLock does not work with system lock

### ISSUE THREAD

**Issue by vit1251** [pixelb/crudini#26]
You will use follow code

``` python
class FileLock(object):
    """Advisory file based locking.  This should be reasonably cross platform
       and also work over distributed file systems."""
    def __init__(self, exclusive=False):
        # In inplace mode, the process must be careful to not close this fp
        # until finished, nor open and close another fp associated with the
        # file.
        self.fp = None
        self.locked = False

        if os.name == 'nt':
            import msvcrt

            def lock(self):
                msvcrt.locking(self.fp.fileno(), msvcrt.LK_LOCK, 1)
                self.locked = True

            def unlock(self):
                if self.locked is True:
                    msvcrt.locking(self.fp.fileno(), msvcrt.LK_UNLCK, 1)
                self.locked = False
```

Also 

``` python
                while True:
                    self.lock()
                    fpnew = os.fdopen(os.open(self.filename, open_mode, 0o666))
                    if os.path.sameopenfile(self.fp.fileno(), fpnew.fileno()):
```

It does not work on WIndows platform!!!

This code have two or three level simplify ready at now...

**Comment by pixelb** [comment]
Do the msvcrt.locking() calls not work?
Does the os.fdopen() call not work?
Have you error messages?

**Comment by vit1251** [comment]
@pixelb you have in self.fp object <FileStream> and use msvcrt.locking(int) where first argument is fileno (file handle).

Second, windows implementation has no sameopenfile implementation.. And as result you tools does not work.

**Comment by pixelb** [comment]
Ah I see tricky. The documentation uses the same 'fd' parameter name for lockf() and locking()

As for the sameopenfile() issue, that shouldn't be possible anyway on windows, so can
you change that part to:

                    if (os.name == 'nt'
                        or os.path.sameopenfile(self.fp.fileno(), fpnew.fileno())):

I'll commit both changes in your name if it works.

thanks!

**Comment by vit1251** [comment]
@pixelb usually I create a separate classes int-t from FileLock for example you may Implement WindowsFileLock and PosixFileLock.

N.B. os.path.sameopenfile you really require this call? You may just read all information to memory and write it again back with exclusive lock. And most of INI files must small size and just load in memory.

**Comment by pixelb** [comment]
Having separate classes seems like a little overkill here. I'll consider cleaning up, though will mean moving the if os == 'nt' call to a higher level.

We need the check to ensure we're locking on the right file. Note the read() is done after the locking.

**Comment by pixelb** [comment]
Addressed in commit 0f2c0b03
